### PR TITLE
Merge-RELEASE. Project 2. #1. Week 12

### PR DESCRIPTION
**Pull request by saffchen**

### CODE REVIEW

File: stock/src/main/java/saffchen/service/StoreService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return storeRepository.findAll();
+    }
+
+    public StoreEntity get(Long id) {
+        return storeRepository.findById(id)
+                              .orElseThrow(() -> new NoEntityException("Object with id " + id + " is not found"));
+    }
+
+    public StoreEntity getByName(String storeName) {
+        return storeRepository.findByName(storeName).get();
```
Comment (pelmegov) [inline stock/src/main/java/saffchen/service/StoreService.java:29]: NPE?

File: stock/src/main/java/saffchen/repository/ProductRepository.java [hunk 2]
```diff
@@ -16,20 +16,20 @@
  */
 @Transactional(readOnly = true)
 @Repository
-public interface ProductsRepository extends BaseRepository<ProductEntity> {
-    @Query(value = "select p from ProductEntity p where p.satellite.id = :satellite")
-    List<ProductEntity> productCountBySatelliteId(@Param("satellite") Long satellite);
+public interface ProductRepository extends BaseRepository<ProductEntity> {
+    /*@Query(value = "select p from ProductEntity p where p.store.id = :storeId")
```
Comment (pelmegov) [inline stock/src/main/java/saffchen/repository/ProductRepository.java:20]: Remove unused code.

File: stock/src/main/java/saffchen/repository/ProductRepository.java [hunk 3]
```diff
@@ -16,20 +16,20 @@
...
+    List<ProductEntity> productCountByStoreId(@Param("store") Long storeId);*/
 
-    @Query("SELECT pr FROM ProductEntity pr WHERE pr.id=:id AND pr.satellite.id=:satelliteId")
-    Optional<ProductEntity> get(Long satelliteId,Long id);
+    @Query("SELECT pr FROM ProductEntity pr WHERE pr.id=:id AND pr.store.id=:storeId")
+    Optional<ProductEntity> get(Long storeId,Long id);
```
Comment (pelmegov) [inline stock/src/main/java/saffchen/repository/ProductRepository.java:24]: Autoformatting.

File: stock/src/main/java/saffchen/mapper/ProductMapper.java [hunk 4]
```diff
@@ -14,7 +14,7 @@
  */
 @Mapper(componentModel = "spring", injectionStrategy = InjectionStrategy.FIELD)
 public interface ProductMapper {
-    @Mapping(source = "satellite.name", target = "satelliteName")
+    @Mapping(source = "store.name", target = "storeName")
```
Comment (pelmegov) [inline stock/src/main/java/saffchen/mapper/ProductMapper.java:17]: Change is supported by frontend?

File: stock/src/main/java/saffchen/entities/ProductEntity.java [hunk 5]
```diff
@@ -60,7 +60,7 @@ public class ProductEntity extends NamedEntity {
     //@JoinColumn(name = "satelliteId", referencedColumnName = "id")
```
Comment (pelmegov) [inline stock/src/main/java/saffchen/entities/ProductEntity.java:60]: Unused code should be deleted.

File: stock/src/main/java/saffchen/config/SecurityConfig.java [hunk 6]
```diff
@@ -25,11 +26,16 @@ protected void configure(HttpSecurity http) throws Exception {
...
-                .mvcMatchers(HttpMethod.PUT,"/api/v1/**").hasAuthority("ROLE_ADMIN")
-                .mvcMatchers(HttpMethod.DELETE,"/api/v1/**").hasAuthority("ROLE_ADMIN")
-                .mvcMatchers(HttpMethod.GET,"/api/v1/**").hasAnyAuthority("ROLE_ADMIN", "ROLE_USER")
+                .requestMatchers(EndpointRequest.to("health")).permitAll()
+                .mvcMatchers(HttpMethod.POST ,"/api/v1/auth/check_auth").anonymous()
+                .mvcMatchers(HttpMethod.GET,"/api/v1/stores/**").permitAll()
+                .mvcMatchers(HttpMethod.GET,"/api/v1/participants").permitAll()
+                .mvcMatchers("/api/v1/admin/**").hasAnyAuthority("ROLE_ADMIN")
+                .mvcMatchers(HttpMethod.POST,"/api/v1/admin/stores/**/products/**").hasAuthority("ROLE_USER")
+                //.mvcMatchers(HttpMethod.PUT,"/api/v1/admin/**").hasAuthority("ROLE_ADMIN")
```
Comment (pelmegov) [inline stock/src/main/java/saffchen/config/SecurityConfig.java:35]: Unused code should be removed.

File: report/src/main/java/saffchen/reports/SwitchTheme.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
+package saffchen.reports;
+
+import com.itextpdf.text.BaseColor;
+
+/**
+ * @author saffchen created on 21.08.2022
+ */
+public class SwitchTheme {
```
Comment (pelmegov) [inline report/src/main/java/saffchen/reports/SwitchTheme.java:8]: This switcher doesn't switch anything, it's better to have one method and switch the theme by requested color.

File: report/src/main/java/saffchen/reports/LightThemePDF.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+    private final Theme theme;
+
+    public LightThemePDF(Theme theme) {
+        this.theme = theme;
+    }
+
+    @Override
+    public BaseColor execute() {
+        return theme.getBaseColor(BaseColor.WHITE);
```
Comment (pelmegov) [inline report/src/main/java/saffchen/reports/LightThemePDF.java:18]: Does it return the same value? Like BaseColor.WHITE? If so why do we need theme here?

**Review by pelmegov — CHANGES_REQUESTED** [review]

File: stock/src/main/java/saffchen/repository/ProductRepository.java (hunk 3 above)
Comment (saffchen) [reply stock/src/main/java/saffchen/repository/ProductRepository.java]: Fixed

File: stock/src/main/java/saffchen/entities/ProductEntity.java (hunk 5 above)
Comment (saffchen) [reply stock/src/main/java/saffchen/entities/ProductEntity.java]: Fixed

File: stock/src/main/java/saffchen/config/SecurityConfig.java (hunk 6 above)
Comment (saffchen) [reply stock/src/main/java/saffchen/config/SecurityConfig.java]: Fixed

File: stock/src/main/java/saffchen/mapper/ProductMapper.java (hunk 4 above)
Comment (aalexeen) [reply stock/src/main/java/saffchen/mapper/ProductMapper.java:17]: I've done it a long time ago. The frontend was working.

File: import/src/main/java/saffchen/command/Command.java [hunk 9]
```diff
@@ -2,5 +2,6 @@
 
 public interface Command {
     String getInfo();
+
```
Comment (aalexeen) [inline import/src/main/java/saffchen/command/Command.java:5]: Just simple refactoring

File: stock/src/main/java/saffchen/repository/ProductRepository.java (hunk 2 above)
Comment (saffchen) [reply stock/src/main/java/saffchen/repository/ProductRepository.java]: Fixed

File: stock/src/main/java/saffchen/service/StoreService.java (hunk 1 above)
Comment (aalexeen) [reply stock/src/main/java/saffchen/service/StoreService.java]: Fixed

**Review by aalexeen — APPROVED** [review]

**Review by pelmegov — APPROVED** [review]

**Pull request merged**